### DISCUSSION TITLE
Any update on batch cotnrolnet images

### DISCUSSION THREAD

**Discussion by demirklvc** [Mikubill/sd-webui-controlnet#459] [General]
### Is there an existing issue for this?

- [X] I have searched the existing issues and checked the recent builds/commits of both this extension and the webui

### What happened?

.

### Steps to reproduce the problem

1. Go to .... 
2. Press ....
3. ...
.

### What should have happened?

.

### Commit where the problem happens

webui: 
controlnet: 

### Command Line Arguments

```Shell
.
```

### Console logs

```Shell
.
```

**Comment by ljleb** [comment]
There is no secretive effort on this as far as I know. Look around a bit in issues and discussion threads, you'll find some conversations on this. If you want this feature to be merged in `main` faster, you should consider participating in discussions imo!

Note: moved to discussions.

**Reply by Petas3** [reply]
Thing is this is faily complex to do to be generally usable - especially given the constraints of the webui. Simple batch is great, but still very far from what you would like to do probably - very far from what I do, running extra steps outside the ui, batch programs etc. Currently I have C# API client that does the batching for me, but its kind of hardwired to do my thing. I might consider upgrading it a bit and releasing, but to provide the UI, its really a lot of extra work.

**Reply by ljleb** [reply]
I agree. With what everyone wants to do, I have a hard time conceptualizing how we would make an orthogonal ui for this. We need to figure out first of all what it is that everyone wants.

So after thinking a bit about this, as I understand it, it seems like people want some kind of movie img2img with precise frame control. Maybe a better place for some of these features would be as an extension in adobe premiere or other video software that calls the webui API? Maybe they should be shoved into another webui extension that uses controlnet? As a new software of its own?

Just some ideas. Putting too many parameters to batch processing could make it harder to maintain in this repo. There is already a lot of things that need to be done. We could focus on fixing existing issues and improve the code that loads and runs the control models if another repo with its own community took care of making an awesome video editor.

**Reply by ljleb** [reply]
Of course maybe I'm just plain wrong and it's a matter of time before we figure it out. Just something that was going in my mind yesterday.

**Reply by demirklvc** [reply]
i would use it to find perfect combinations. its a pain to reload several times images. it would help the worfklow if you could run a whole folder of control net images and later compare which imagery worked best.(with a fixed seed) this would also help the learning process.

**Reply by demirklvc** [reply]
im not looking for an video option/video editor. jsut want to be able to work faster and better

**Reply by ljleb** [reply]
You are not looking for that, and I totally understand. It looked to me like some propositions were going in that direction though. See https://github.com/Mikubill/sd-webui-controlnet/issues/243#issuecomment-1447603112 for example.

Not everyone seem to want the same thing, which is why I had this feeling that there might be too many features to put in to satisfy everybody.

**Reply by demirklvc** [reply]
I mean, my hope for an batch processing isnt anything experimental. Rather a standard feature i would say.

**Reply by Petas3** [reply]
Yes agree the simple batch would be very helpful for most users.
Edit: Just keep in mind 'with food comes the hunger' and once you develop a taste for this nothing will be enough 🍗 consider trying the API, although its a bit broken atm.

**Reply by ljleb** [reply]
I agree that normal batch with controlnet would be quite useful. All I'm saying is maybe there's a better place for advanced features like precise frame control.
